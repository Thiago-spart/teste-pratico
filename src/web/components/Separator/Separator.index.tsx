import type { SeparatorProps } from "./Separator.types";

import * as S from "./Separator.styled";

export const Separator: React.FC<SeparatorProps> = ({ children }) => {
	return (
		<S.Container>
			<span>{children}</span>
		</S.Container>
	);
};
