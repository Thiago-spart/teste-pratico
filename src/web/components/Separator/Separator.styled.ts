import styled from "styled-components";

export const Container = styled.div`
	display: flex;
	align-items: center;
	justify-content: center;
	font-weight: 500;
	width: 100%;
	position: relative;

	&::before,
	&::after {
		position: absolute;
		width: 45%;
		height: 2px;
		background-color: #e0e0e0;
		content: "";
	}

	&::before {
		right: 0;
	}

	&::after {
		left: 0;
	}
`;
