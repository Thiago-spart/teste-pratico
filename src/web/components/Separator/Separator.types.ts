import type { ReactNode } from "react";

export interface SeparatorProps {
	children: ReactNode;
}
