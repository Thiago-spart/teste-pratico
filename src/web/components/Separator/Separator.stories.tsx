/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import { Separator } from "./Separator.index";

const meta: Meta<typeof Separator> = {
	title: "Componentes/Separator",
	component: Separator,
};

const Template: ComponentStory<typeof Separator> = props => (
	<Separator {...props} />
);

export const Default = Template.bind({});
Default.args = {
	children: <span>ou</span>,
};

export default meta;
