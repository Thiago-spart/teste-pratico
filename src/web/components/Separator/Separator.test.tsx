/* eslint-disable sonarjs/no-duplicate-string */
import { composeStories } from "@storybook/testing-react";
import { render } from "@testing-library/react";

import * as stories from "./Separator.stories";

const { Default } = composeStories(stories);

describe("Button component", () => {
	it("should match snapshot", () => {
		const { container } = render(<Default />);

		expect(container).toMatchSnapshot();
	});
});
