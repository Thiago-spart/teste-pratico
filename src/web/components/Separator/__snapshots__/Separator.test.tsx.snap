// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ bKioOW"
  >
    <span>
      <span>
        ou
      </span>
    </span>
  </div>
</div>
`;
